napari/layers/points/_tests/test_points.py

```diff
@@ -2502,3 +2488,34 @@ def test_set_drag_start():
     np.testing.assert_array_equal(layer._drag_start, position)
     layer._set_drag_start({0}, position=(1, 2))
     np.testing.assert_array_equal(layer._drag_start, position)
+
+
+def test_point_slice_request_response():
+    """Test points slicing with request and response."""
+    data = [
+        (10, 2, 4),
+        (10 + 2 * 1e-7, 4, 6),
+        (8, 1, 7),
+        (10.1, 7, 2),
+        (10 - 2 * 1e-7, 1, 6),
+    ]
+
+    layer = Points(data)
+
+    dims_indices_list = [
+        (8, slice(None), slice(None)),
+        (10, slice(None), slice(None)),
+        (10 + 2 * 1e-12, slice(None), slice(None)),
+        (10.1, slice(None), slice(None)),
+    ]
+    length_check_list = [1, 4, 4, 4]
+
+    for dims_indices, length_check in zip(
+        dims_indices_list, length_check_list
+    ):
```

Comment:
Any reason not to use [`pytest.mark.parametrize`](https://docs.pytest.org/en/6.2.x/parametrize.html) here? It should make specific cases failing easier to read.